### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by inakiesk**
This branch is dedicated to adding Jest testing to the Math magicians project.

The features are the following:

- Write unit tests for operate.js and calculate.js.
- Used Jest snapshots to test the rest of the components.
- Used React testing library.
- All tests are passing.

### CODE REVIEW

File: src/components/logic/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      next: null,
+      operation: '+',
+    };
+
+    const buttonName = '5';
+
+    const result = calculate(obj, buttonName);
+
+    expect(result).toEqual({ next: '5', operation: '+', total: '10' });
+  });
```
Comment (Gambit142) [inline src/components/logic/calculate.test.js:16]: - [ ] Great work writing unit tests for the `calculate` function. Based on the requirement, you should test all functionalities and scenarios for the `calculate` function. Kindly add unit tests that cover the following scenarios:

- When the `buttonName === 'AC'`
- When `buttonName === '.'`
- When `buttonName === '='`
- When `buttonName === '+/-'` 💻 👍

File: src/components/logic/operate.test.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  test('test if total is correct', () => {
+    const numberOne = '5';
+    const numberTwo = '6';
+    const operation = '+';
+
+    const result = operate(numberOne, numberTwo, operation);
+
+    expect(result).toEqual('11');
+  });
+});
```
Comment (Gambit142) [inline src/components/logic/operate.test.js:13]: - [ ] Great work writing unit tests for the scenario when the operation === '+' for the `operate` function. Based on the requirement, you should test all functionalities and scenarios for the `operate` function. Kindly add unit tests that cover the following scenarios:

- When `operation === '-'`
- When `operation === 'x'`
- When `operation === '÷'`
- When `operation === '%'` 👍 💻

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @inakiesk @Gdiazdiaz 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Jest library was used to  write unit tests for functions ✔️ 
- Jest _**snapshot**_ was used to test React components ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/logic/calculate.test.js (hunk 1 above)
Comment (rloterh) [reply src/components/logic/calculate.test.js]: well resolved! ✔️

File: src/components/logic/operate.test.js (hunk 2 above)
Comment (rloterh) [reply src/components/logic/operate.test.js]: well resolved! ✔️

File: src/components/Navbar.test.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+import renderer from 'react-test-renderer';
+import { BrowserRouter as Router } from 'react-router-dom';
+import Navbar from './Navbar';
+
+describe('Checking page deployment', () => {
+  test('Navbar component renders', () => {
+    const tree = renderer.create(<Router><Navbar /></Router>).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
```
[the remark this answers is not in the document]
Comment (rloterh) [reply src/components/Navbar.test.js]: well resolved! ✔️

File: src/components/Calculator.test.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Calculator from './Calculator';
+
+describe('Checking calculator deployment', () => {
+  test('Calculator renders', () => {
+    const tree = renderer.create(<Calculator />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
```
[the remark this answers is not in the document]
Comment (rloterh) [reply src/components/Calculator.test.js]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Inaki & Glen,  

Your project is complete! 🏁 Great job! There is nothing else to say other than... it's time to merge it 🍾🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
- Good unit test coverage using _Jest_!
- Nice job applying the requested changes!

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
- Kindly include details of the other co-contributor(s) in your ReadMe.

Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**